### PR TITLE
[CM-1449] Dark Mode Support part -1 | iOS SDK

### PR DESCRIPTION
**Pull request by AbhijeetRanjan308**
## Summary
- Exposed a method to enable the Dark Mode (.isDarkModeEnabled) from default configuration. (by default its false)
- Conversation Screen
- Conversation List Screen
- Navigation Bar 
- ChatBar 
- typingText Color dynamic Update
- Suggested Button 
- Received Message text and Bubble dynamic update
- Form Templets text and view dynamic update 
- Custom Picker color dynamic update
- Auto Suggestion View
- Generic Card Rich Message
- DropDown list ( Partially done: dropdown icon is not changing color dynamically ) 
- Card Carousel dynamic update 
- Fixed Document Sending UI issue.

## Reference
- Figma link : https://www.figma.com/file/4cSLo3RwfBgZCZCeTtKzgP/iOS-UI%2FUX?type=design&node-id=0%3A1&mode=design&t=SIX3kccGAY0KhEtJ-1

## Video Preview

https://github.com/Kommunicate-io/KommunicateChatUI-iOS-SDK/assets/139110221/fa8ed0cc-716d-4c83-8f75-d560f5bcd5f0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 The changelog for [KommunicateChatUI-iOS-SDK](https://github.com/Kommunicate-io/KommunicateChatUI-iOS-SDK). Also see the [releases](https://github.com/Kommunicate-io/KommunicateChatUI-iOS-SDK/releases) on Github.
 
+## Unreleased
```
Comment (Sathyan-KM) [inline CHANGELOG.md:5]: Remove this. We can add this once we complete the dark mode support

File: Sources/Controllers/ALKConversationViewController.swift [hunk 2]
```diff
@@ -410,6 +410,14 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
...
 
+    func switchDynamicMode(isDynamic: Bool) {
+        if isDynamic {
+                UIApplication.shared.windows.first?.overrideUserInterfaceStyle = .unspecified
+        } else {
+                UIApplication.shared.windows.first?.overrideUserInterfaceStyle = .light
+        }
+    }
+    
     open func updateAssigneeOnlineStatus(userId: String){}
```
Comment (Sathyan-KM) [inline Sources/Controllers/ALKConversationViewController.swift:421]: its is used in different view controller and repeated. Can we add this into VC extension & use in all places ?

File: Sources/Controllers/ALKConversationViewController.swift [hunk 3]
```diff
@@ -796,11 +804,12 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
     // swiftlint:disable:next cyclomatic_complexity function_body_length
     private func prepareChatBar() {
         // Update ChatBar's top view which contains send button and the text view.
-        chatBar.grayView.backgroundColor = configuration.backgroundColor
+        chatBar.grayView.backgroundColor = UIColor.dynamicColor(light: configuration.backgroundColor, dark: UIColor(netHex: 0x1C1C1C))
 
         // Update background view's color which contains all the attachment options.
-        chatBar.bottomBackgroundColor = configuration.chatBarAttachmentViewBackgroundColor
+        chatBar.bottomBackgroundColor = UIColor.dynamicColor(light: configuration.chatBarAttachmentViewBackgroundColor, dark: UIColor(netHex: 0x313131))
```
Comment (Sathyan-KM) [inline Sources/Controllers/ALKConversationViewController.swift:810]: 0x1C1C1C & 0x313131 these also used different places & hardcoded in everyplace. instead we can make it constant  single place & use it in all places

File: Sources/Controllers/ALKMultipleLanguageSelectionViewController.swift [hunk 4]
```diff
@@ -61,7 +61,7 @@ class ALKMultipleLanguageSelectionViewController : UIViewController {
         super.viewDidLoad()
         if #available(iOS 13.0, *) {
             // Always adopt a light interface style.
-            overrideUserInterfaceStyle = .light
+//            overrideUserInterfaceStyle = .light
```
Comment (Sathyan-KM) [inline Sources/Controllers/ALKMultipleLanguageSelectionViewController.swift:64]: we can remove this right ?

File: Sources/Models/ALKConfiguration.swift [hunk 5]
```diff
@@ -73,6 +73,9 @@ public struct ALKConfiguration {
 
     /// Additional information you can pass in message metadata in all the messages.
     public var messageMetadata: [AnyHashable: Any]?
+    
+    /// Chcek for the dark mode Support Enabled for the SDK
```
Comment (Sathyan-KM) [inline Sources/Models/ALKConfiguration.swift:77]: Write the description clearly & without typo

File: Sources/Views/ALKDocumentCell.swift [hunk 6]
```diff
@@ -92,7 +93,7 @@ class ALKDocumentCell: ALKChatBaseCell<ALKMessageViewModel> {
         label.numberOfLines = 1
         label.font = UIFont.systemFont(ofSize: 12)
         label.isOpaque = true
-        label.textColor = .black
+        label.textColor = .white
```
Comment (Sathyan-KM) [inline Sources/Views/ALKDocumentCell.swift:96]: hardcoded colors will be same for both modes?

**Review by Sathyan-KM — CHANGES_REQUESTED** [review]

File: Sources/Views/ALKDocumentCell.swift (hunk 6 above)
Comment (AbhijeetRanjan308) [reply Sources/Views/ALKDocumentCell.swift:96]: As we are not modifying the message sending process, it won't require customization for dark mode support.

File: CHANGELOG.md (hunk 1 above)
Comment (AbhijeetRanjan308) [reply CHANGELOG.md]: done

File: Sources/Controllers/ALKConversationViewController.swift (hunk 2 above)
Comment (AbhijeetRanjan308) [reply Sources/Controllers/ALKConversationViewController.swift:421]: moved it to UiViewContoller extension.

File: Sources/Controllers/ALKConversationViewController.swift (hunk 3 above)
Comment (AbhijeetRanjan308) [reply Sources/Controllers/ALKConversationViewController.swift]: done

File: Sources/Controllers/ALKMultipleLanguageSelectionViewController.swift (hunk 4 above)
Comment (AbhijeetRanjan308) [reply Sources/Controllers/ALKMultipleLanguageSelectionViewController.swift]: removed this

File: Sources/Models/ALKConfiguration.swift [hunk 7]
```diff
@@ -73,6 +73,9 @@ public struct ALKConfiguration {
 
     /// Additional information you can pass in message metadata in all the messages.
     public var messageMetadata: [AnyHashable: Any]?
+    
+    /// Indicates whether dark mode support is enabled for the SDK.
```
[the remark this answers is not in the document]
Comment (AbhijeetRanjan308) [reply Sources/Models/ALKConfiguration.swift:77]: okay

**Pull request merged**